### ISSUE TITLE
Crashing when trying to organize files...

### ISSUE THREAD

**Issue by Ronald-Diemicke** [trembon/switch-library-manager#79]
So - it's strange.... I've got a number of updates I'm trying to organize and the application just dies in windows. I turned on debugged but there doesn't seem to be anything useful... It's happening with a small subset... the strange part is that for other files, it works just fine...

These are some of the files its dying with:
Baten Kaitos I and II HD Remaster Upd v1.0.3 [UPD][0100f28018ca4800][v196608].nsp Dariusburst Another Chronicle EX Plus Upd v1.0.3 [UPD][010015800f93c800][v196608].nsp
Demon Throttle Upd v1.0.1 [UPD][01001da015650800][v65536].nsp
Double Dragon Gaiden Rise of the Dragons Upd v1.0.6 [UPD][010010401bc1a800][v327680].nsp
Dungreed Upd v1.7.6.0 [UPD][010045b00c496800][v589824].nsp
Emio The Smiling Man Famicom Detective Club Upd v1.1.0 [UPD][010054601d54c800][v65536].nsp
FREDERICA Upd v1.0.3 [UPD][0100bae01a078800][v196608].nsp

I tried renaming them and that didn't help either... Any ideas?

**Comment by trembon** [comment]
it is possible they have made changes to the structure in the files with the latest versions of the firmware, will try to see if i can replicate the error with any of the files you mentioned here

**Comment by Ronald-Diemicke** [comment]
Yeah - I suspect it's something like that; it might be worth seeing if the section of the code that is accessing those bits in the files can be made to fail more gracefully or output errors to the log... I might try to take a look at the code later, but that's what I'd do first!

**Comment by vondutch2560** [comment]
I haven’t encountered any crashes while organizing files—everything works perfectly for me. However, I have a bit of OCD when it comes to organization. Would it be possible to add "BASE" as an additional option under {TYPE}, alongside "DLC" and "UPD"? This would make it much clearer for me to determine the installation order and would make me very happy. Thank you!

**Comment by trembon** [comment]
@vondutch2560 shouldnt be that big of a change, but i think it will be needed as an option to enable it, but can you create it as a separate issue rather then a comment here?

**Comment by s1rleblanc** [comment]
> So - it's strange.... I've got a number of updates I'm trying to organize and the application just dies in windows. I turned on debugged but there doesn't seem to be anything useful... It's happening with a small subset... the strange part is that for other files, it works just fine...
> 
> These are some of the files its dying with: Baten Kaitos I and II HD Remaster Upd v1.0.3 [UPD][0100f28018ca4800][v196608].nsp Dariusburst Another Chronicle EX Plus Upd v1.0.3 [UPD][010015800f93c800][v196608].nsp Demon Throttle Upd v1.0.1 [UPD][01001da015650800][v65536].nsp Double Dragon Gaiden Rise of the Dragons Upd v1.0.6 [UPD][010010401bc1a800][v327680].nsp Dungreed Upd v1.7.6.0 [UPD][010045b00c496800][v589824].nsp Emio The Smiling Man Famicom Detective Club Upd v1.1.0 [UPD][010054601d54c800][v65536].nsp FREDERICA Upd v1.0.3 [UPD][0100bae01a078800][v196608].nsp
> 
> I tried renaming them and that didn't help either... Any ideas?

I have same issue trying to organize and app dies.. im trying at macOS and this is the result of my research

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x138e794]
--------------------------------
null in function getTitleName

github.com/trembon/switch-library-manager/process.getTitleName
/home/runner/work/switch-library-manager/switch-library-manager/src/process/organizefolderStructure.go:328
---------------------------------

here seems like everything goin good until panic

2025/01/31 00:08:19 Sending to Astilectron: {"name":"window.cmd.message","targetID":"1","callbackId":"35",...}
2025/01/31 00:08:19 Sending to Astilectron: {"name":"window.cmd.message","targetID":"1","callbackId":"36",...}
panic: runtime error: invalid memory address or nil pointer dereference

error in here /home/runner/work/switch-library-manager/switch-library-manager/src/gui.go:430


As a final comment, the directories and folders are not mine, they belong to the programmer and the program is trying to access those addresses.

**Comment by s1rleblanc** [comment]
ok, I don't know if is correct put this in here but.. I fix the problem.

the issue is in src/process/organizefolderStructure.go

in the function getTitleName

I made this changes

```
func getTitleName(switchTitle *db.SwitchTitle, v *db.SwitchGameFiles) string {
    // Check if switchTitle is not nil and contains a name
    if switchTitle != nil && switchTitle.Attributes.Name != "" {
        res := cjk.FindAllString(switchTitle.Attributes.Name, -1)
        if len(res) == 0 {
            return switchTitle.Attributes.Name
        }
    }

    // Check if v and its Metadata are present
    if v != nil && v.File.Metadata != nil && v.File.Metadata.Ncap != nil {
        // Check if the title name exists
        name := v.File.Metadata.Ncap.TitleName["AmericanEnglish"].Title
        if name != "" {
            return name
        }
    }

    // For non-eShop games, get the name from the file
    if v != nil {
        return db.ParseTitleNameFromFileName(v.File.ExtendedInfo.FileName)
    }

    // Default return if no valid name is found
    return "Unknown Title"
}
```

then compile it and it works!

**Comment by Ronald-Diemicke** [comment]
@s1rleblanc maybe branch or fork and put in a PR for @trembon ?

**Comment by trembon** [comment]
a PR would be nice, but i will try to move it in later otherwise, but my focus has been elsewhere the last few month so i havent really had the energy to code on this project, but its not forgotten :)

**Comment by trembon** [comment]
added provided fix and should now be in the latest beta build available under the Actions tab

**Comment by emumovies** [comment]
I notice it only crashes in 1.9.0 when I set "process_when_missing_base_game": true 
If I set it to false the app seems rock solid.

**Comment by SteveMyers75** [comment]
> I notice it only crashes in 1.9.0 when I set "process_when_missing_base_game": true
> If I set it to false the app seems rock solid.

Same here. 

Works great except when the "**_process_when_missing_base_game_**" parameter is set to **True**.
Has anyone figured a workaround?  
_(besides setting that feature to false... it is a really useful feature!)_

**Comment by SteveMyers75** [comment]
Update to the crashing when **process_when_missing_base_game** parameter is set to True.

I found at least one update that causes it to crash every time I try to process it.
If I remove it, SLM19 runs perfectly and renames the files without any issue.
As soon as I add this one update, it crashes. (_I am an Airtraffic Controller[01002EE01BAE0800][589824].nsp_) 

I verified that the update is valid and installs fine.
I can email it to someone so they can figure out why some updates cause SLM19 to crash if that helps.

**Note: This is not the only file that causes SLM19 to crash, just a good example that might help solve the problem**

**Comment by trembon** [comment]
@SteveMyers75 thanks for the feedback, will see if i can replicate it with this information when i have time to work on it :)
